js/utils.js

```diff
@@ -1106,19 +1107,70 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
     }
 
     return _.zipObject(fields, _.map(fields, function(field) {
-      return _.sortBy(_.uniq(splitByCommas(_.map(records, 'data.' + field))));
+      return _.sortBy(_.uniq(splitByCommas(_.map(records, ['data', field]))));
     }));
   }
 
+  /**
+   * Takes the list of data records and the page query parameter to return a list of filter and filter options to render
+   * @param {Object} options - A mapping of options for this function
+   * @param {Object[]} options.records - Records from which to derive the filter values
+   * @param {Object[]} options.filters - List of filters configured by the user
+   * @param {Number} options.id - Widget instance ID
+   * @param {Object} [options.query] - Filter query found in the page URL
+   * @returns {Object[]} An array of filters and possible values, ready to be rendered through Handlebars template
+   */
   function parseRecordFilters(options) {
     options = options || {};
 
     var records = options.records || [];
     var filters = options.filters || [];
     var id = options.id;
+    var query = options.query;
+    var hasFilterQuery = query && query.value.length;
+
+    // Add a fake entry into records to represent the filters from the query
+    // to ensure all filter options from the query are rendered in the filter UI
+    if (hasFilterQuery) {
+      var flFilters = [];
+
+      // When filter columns are unspecified, apply the values to all columns
+      if (!query.column.length) {
+        query.column = filters;
+        query.value = _.fill(Array(filters.length), query.value);
+      }
+
+      _.forEach(query.column, function(field, i) {
+        var value = query.value[i];
+
+        if (!Array.isArray(value)) {
+          value = [value];
+        }
+
+        _.forEach(value, function(value) {
+          if (typeof value === 'undefined') {
+            return;
+          }
+
+          flFilters.push({
+            type: field,
+            data: {
+              name: value,
+              class: ('' + value).toLowerCase().replace(/[!@#\$%\^\&*\)\(\ ]/g, '-')
```

Comment:
What's this regexp for? Leave a comment to explain what it does and matches

Reply:
@squallstar This was for creating kebab case for class names. I've upgraded this to use `_.kebabCase()`.